Reflection/Interfaces.simba

```diff
@@ -0,0 +1,199 @@
+{$include_once Internal/Reflection.simba}
+{$include_once Constants.simba}
+{$include_once Menu.simba}
+{$include_once Inventory.simba}
+{$include_once Mouse.simba}
+{$include_once Timing.simba}
+
+Function R_WidgetIsHidden(Constant: RWidget): Boolean;
+var
+  Widget: RSWidget;
+begin
+  Widget := RSWidget.Get(Constant.Group, Constant.Child);
+  if Widget.ref = nil then
+    Exit(False);
+
+  Result := Widget.IsHidden;
+  Widget.Free;
+end;
+
+(*
+MAGIC SPELL BOOK
+~~~~~~~~~~~~~~~~
+
+Valid spells are located in Constants.simba
+  Ex: R_SPELL_VARROCK_TELEPORT
+
+*)
+
+Function R_SpellBook: Boolean;
+begin
+  Result := RSWidget.IsValid(R_SPELLBOOK_CONTAINER.Group, R_SPELLBOOK_CONTAINER.Child);
+end;
+
+Function R_WaitSpellBook(timeout: UInt32): Boolean;
+begin
+  Result := R_WaitFunction(timeout, 50, @R_SpellBook);
+end;
+
+Function R_CastSpell(Spell: RWidget): Boolean;
```

Comment:
Paramenters to `R_CastSpell` should just be an enum :)
Check out how we did the `GameTab.simba` file.

That way, you would just have one `RWidget` representing the parent interface, and the enum values would be the child or group so that if stuff breaks, you don't have to change a ton of widget constants :D but just the enums in that one file.

I can fix that myself if you like (after merging).

Reply:
I am currently converting these and will push the changes when I am done :)